addon/components/uni-autocomplete.js

```diff
@@ -35,19 +38,38 @@ export default Component.extend(ClickOutside, {
 
       let options = this.get('options').map((option) => {
         let options = this.get('searchTextValues')(option).map((x) => x.toLowerCase());
-        let matchedValues = options.filter((el) => el.startsWith(this.get('valueLowerCase')));
-
+        let matchedValues = A(options.filter((el) => el.startsWith(this.get('valueLowerCase'))));
         return { option, matchedValues };
       });
 
-      return options.filter(({ matchedValues }) => isPresent(matchedValues)).slice(0, this.get('maxOptionsToShow'));
+      return A(options.filter(({ matchedValues }) => isPresent(matchedValues)).slice(0, this.get('maxOptionsToShow')));
     },
 
     set(_, value) {
       return value;
     }
   }),
 
+  autocompleteOption: computed('optionsFiltered', 'highlighted', function() {
+    let options = this.get('optionsFiltered');
+
+    return isEmpty(options) ? null : options.objectAt(this.get('highlighted'));
+  }),
+
+  autocompleteText: computed('autocompleteOption', 'value', function() {
+    let option = this.get('autocompleteOption');
+
+    if (isEmpty(option)) {
+      return '';
```

Comment:
return placeholder?

Reply:
nop, this is an autofill for when we are wrinting. No need to return the placeholder here.